### PR TITLE
Enhancements to Docs

### PR DESCRIPTION
**Pull request by martinpeck**
This PR performs some tidy up of the docs, updates some stuff that was out of date, and attempts to make the docs "flow" more naturally for anyone picking up the repo for the first time.

There's more that could be done here, but the things that have changed include:

1. Add some additional context setting and explanation to the README
2. Add a structured "how to get started" to the README
3. Use the name "Azure OpenAI API Simulator" more often (and especially in doc titles)
4. Remove info about "running the simulator" from `config.md` and move that into `running-deploying.md`
5. Expand on how to work with .env file in `config.md`
6. Enhance `developing.md` to describe the main `make` rules, and fix up the description of the folders (it was out of date)
7. Minor changes (title) to `extending.md`
8. Add instructions on running locally to `running-deploying.md` and include notes about the mode that were previously in `config.md`

I'm sure there are other enhancements to be made, but I feel this is a good start and will help people onboard to the project.

Addresses #23

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,49 +39,60 @@ Let's assume that you have build a chatbot that uses the OpenAI API to generate
...
-- **Integration Testing**: The simulator can be used to run integration tests, for example in CI builds without needing to have credentials for an Azure OpenAI endpoint
+## How to Get Started with the Azure OpenAI API Simulator
 
-The simulator is not a replacement for testing against the real Azure OpenAI API, but it can be a useful tool in your testing toolbox.
+### Running and Deploying the Azure OpenAI API Simulator
+To document [Running and Deploying the Azure OpenAI API Simulator](./docs/running-deploying.md) includes instructions on running the Azure OpenAI API Simulator locally, packaging and deploying it in a Docker container, and also deploying the Azure OpenAI API Simulator to Azure Container Apps.
 
-## Read More
+### Configuring the Azure OpenAI API Simulator
+The behavious of the Azure OpenAI API Simulator is controlled via a range of [Azure OpenAI API Simulator Configuration Options](./docs/config.md).
```
Comment (stuartleeks) [inline README.md:88]:
```suggestion
The behaviour of the Azure OpenAI API Simulator is controlled via a range of [Azure OpenAI API Simulator Configuration Options](./docs/config.md).
```

File: README.md [hunk 2]
```diff
@@ -35,49 +39,60 @@ Let's assume that you have build a chatbot that uses the OpenAI API to generate
...
 
-There are various options for [running and deploying](./docs/running-deploying.md) the simulator, including running in Docker and deploying to Azure Container Apps.
+### Extending the Azure OpenAI API Simulator
+There are also a number of [Azure OpenAI API Simulator Extensions](./docs/extensions.md) that allow you to customise the behaviour of the Azure OpenAI API Simulator. Extensions can be used to modify the request/response, add latency, or even generate responses.
```
Comment (stuartleeks) [inline README.md:91]:
```suggestion
There are also a number of [Azure OpenAI API Simulator Extension points](./docs/extensions.md) that allow you to customise the behaviour of the Azure OpenAI API Simulator. Extensions can be used to modify the request/response, add latency, or even generate responses.
```

File: docs/running-deploying.md [hunk 3]
```diff
@@ -1,20 +1,62 @@
...
     - [Unrestricted Network Access](#unrestricted-network-access)
     - [Semi-Restricted Network Access](#semi-restricted-network-access)
     - [Restricted Network Access](#restricted-network-access)
+  - [Managing Large Recordings](#managing-large-recordings)
 
 
 ## Getting Started
 
-This repo is configured with a Visual Studio Code [dev container](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) that sets up a Python environment ready to work in.
+The simplest way to work with and the simulator code is within a [Dev Containers](https://code.visualstudio.com/docs/devcontainers/containers) in VS Code.
```
Comment (stuartleeks) [inline docs/running-deploying.md:18]:
```suggestion
The simplest way to work with and the simulator code is within a [Dev Container](https://code.visualstudio.com/docs/devcontainers/containers) in VS Code.
```

**Review by stuartleeks — APPROVED** [review]
Left a few comments for minor tweaks but other than that, LGTM

Thanks!

**Comment by martinpeck** [comment]
Have accepted and merged all of your suggestion.
Thanks for reviewing @stuartleeks!

**Pull request merged**